### ISSUE TITLE
[Feature Request] Pydantic Support for Request body, Response & Query Parameters

### ISSUE THREAD

**Issue by omfd** [axnsan12/drf-yasg#734]
Hi Guys,

Pydantic by default provide support for schema generation from the Data Model. If someone can create a feature branch for support pydantic objects as well for the above mentioned things -  it will be awesome !! 

currently it works for drf serializers only - 

just a request - might prove helpful for a quite a lot of developers !! 

Regards

**Comment by antonagestam** [comment]
@omfd I'm curious if anyone has experimented with making this work, e.g. using some transformer from pydantic's `BaseModel.dict()` to drf-yasg's `Schema`?

**Comment by antonagestam** [comment]
@JoelLefkowitz I think it would make sense to use the same facilities as are already in place. Allowing pydantic models in the `responses` and `request_body` arguments of `swagger_auto_schema`, e.g. `@swagger_auto_schema(request_body=SomeRequest, responses={200: SomeResponse})`. I think that would solve 95% of cases :)

**Comment by omfd** [comment]
@antonagestam exactly! If you have a look at the Fastapi sample I am mentioning below for reference ! 

`class Item(BaseModel):
    name: str
    description: Optional[str] = None
    price: float
    tax: Optional[float] = None


@app.post("/items/", response_model=Item)
async def create_item(item: Item):
    return item`

The above is a POST request with Body as of type **Item** and response is also of the same type **Item**

Below is the screenshot for the swagger generator Fastapi guys have done for themselves where default if Pydantic Data Models

![Screenshot_2021-10-16_14-42-36](https://user-images.githubusercontent.com/25281663/137581894-af391d32-1f66-464f-bd33-09edbb62ed86.png)

If we add support for adding **request_body** and **responses** object derived from Pydantic BaseModels it will be pretty good.

Regards

**Comment by omfd** [comment]
> @omfd I'm curious if anyone has experimented with making this work, e.g. using some transformer from pydantic's `BaseModel.dict()` to drf-yasg's `Schema`?

@antonagestam I tried to extend SwaggerAutoSchema to accomodate support for the pydantic objects. 

But the prime issue is the base class has a strict check of object being of type serializer which makes it incompatible 


Even if that base method can be released with support of Union[Serializer, BaseModel]

we can customize it and extend it accordingly as per the need but the above support needs to be there !! 

Rest pydantic by default have pretty good method to make it work once that strict check and include pydantic object support

**Comment by antonagestam** [comment]
@omfd 

> But the prime issue is the base class has a strict check of object being of type serializer which makes it incompatible

Which check is that? Is it in `BaseInspector`?

**Comment by Utsavan** [comment]
Has anyone tried using already provided  ```schema()``` method in ```pydantic```  with already provided ```openapi.Schema``` class in ```drf_yasg``` to generate the docs, something like this -- 
```
class Item(BaseModel):
    name: str
    description: Optional[str] = None
    price: float
    tax: Optional[float] = None


from drf_yasg.openapi import Schema
from drf_yasg.utils import swagger_auto_schema
@swagger_auto_schema(
        responses={
            status.HTTP_200_OK: Schema(**Item.schema())
        }
    )
```

**Comment by utapyngo** [comment]
> Has anyone tried using already provided `schema()` method in `pydantic` with already provided `openapi.Schema` class in `drf_yasg` to generate the docs, something like this --

This does not seem to work with nested models. I am getting
```
drf_yasg.errors.SwaggerValidationError: spec validation failed: 
{'ssv': '("Unresolvable JSON pointer: \'definitions/SubModel\'")'}
```

**Comment by jclerman** [comment]
Consider using the approach here (https://github.com/pydantic/pydantic/issues/889#issuecomment-1064688675) to avoid submodels in the JSON schema generated by pydantic.
